### ISSUE TITLE
Set XMLHttpRequest "withCredentials" attribute for CORS

### ISSUE THREAD

**Issue by rachelappel** [aspnet/Docs#6078]
See this thread, we need to document this for the 2.2 release.
https://github.com/aspnet/SignalR/issues/2110

**Comment by anurse** [comment]
This issue occurs in the 2.1 release and numerous customers are hitting it. I think we should up the priority of this. Also, we should document that the Azure App Service CORS settings are **not** compatible with SignalR, and you should only use the ASP.NET Core CORS functionality.

**Comment by anurse** [comment]
I'm re-titling this issue to cover the main problem. We need to have an article on configuring CORS support for ASP.NET Core SignalR. A number of customers are seeing issues with this and our docs have [some content](https://docs.microsoft.com/en-us/aspnet/core/signalr/get-started?view=aspnetcore-2.1&tabs=visual-studio#configure-the-project-to-use-signalr) but are a little light on the subject.

Outline:

* CORS is needed when your JavaScript client is hosted on a different domain than your SignalR Hubs
* SignalR **requires** the `.AllowCredentials` option be specified on the CORS policy on the server
* SignalR uses the `GET` and `POST` methods (so you can filter to only those methods, or just use `.AllowAnyHeader()`
* Sample errors on the browser that indicate CORS isn't working:
   * `No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'null' is therefore not allowed access.`

**Comment by rachelappel** [comment]
Good point @anurse. The docs assume you know what CORS is and doesn't go into detail, just shows how to do config in Startup.

**Comment by anurse** [comment]
I'd suggest P1

**Comment by virtualbiz** [comment]
![image](https://user-images.githubusercontent.com/50282257/59648898-3708ae00-91b3-11e9-8932-6fb2ffd3d51d.png)
Somehow I have to use this combination of configuration to make it work. Missing either AllowCredentials or SetIsOriginAllowed won't work. I think the document needs to provide more details and sample code.
